ts/screens/onboarding/EmailInsertScreen.tsx

```diff
@@ -0,0 +1,201 @@
+   * - _true_,      if email is valid.
+   * - _false_,     if email has been already changed from the user and it is not
+   * valid.
+   * - _undefined_, if email field is empty. This state is consumed by
+   * LabelledItem Component and it used for style pourposes ONLY.
+   */
+  private isValidEmail() {
+    return this.state.email
+      .map(value => {
+        return EmailString.decode(value).isRight();
+      })
+      .toUndefined();
+  }
+
+  private updateEmailState(value: string) {
+    this.setState({
+      email: value !== EMPTY_EMAIL ? some(value) : none
+    });
+  }
+
+  public render() {
+    return (
+      <BaseScreenComponent
+        goBack={this.handleGoBack}
+        headerTitle={I18n.t("onboarding.email.insert.headerTitle")}
+        contextualHelp={{
+          title: I18n.t("onboarding.email.insert.help.title"),
+          body: () => (
+            <Markdown>
+              {I18n.t("onboarding.email.insert.help.content")}
+            </Markdown>
+          )
+        }}
+      >
+        <View style={styles.container}>
+          <Content noPadded={true} style={styles.content} scrollEnabled={false}>
+            <H4 style={[styles.boldH4, styles.horizontalPadding]}>
+              {I18n.t("onboarding.email.insert.title")}
+            </H4>
+            <View spacer={true} />
+            <View style={styles.horizontalPadding}>
+              <Text>{I18n.t("onboarding.email.subtitle")}</Text>
+            </View>
+            <View spacer={true} />
+            <View style={styles.horizontalPadding}>
+              <Form>
+                <LabelledItem
+                  type={"text"}
+                  label={I18n.t("onboarding.email.emailInputLabel")}
+                  icon="io-envelope"
+                  isValid={this.isValidEmail()}
+                  inputProps={{
+                    autoCapitalize: "none",
+                    value: this.state.email.getOrElse(EMPTY_EMAIL),
+                    onChangeText: (value: string) =>
+                      this.updateEmailState(value),
+                    style: styles.emailInput
+                  }}
+                  iconStyle={styles.icon}
+                />
+              </Form>
+            </View>
+          </Content>
+
+          <KeyboardAvoidingView
+            behavior="padding"
+            keyboardVerticalOffset={Platform.select({
+              ios: 100,
+              android: 80
+            })}
+          >
+            {this.renderFooterButtons()}
+          </KeyboardAvoidingView>
+        </View>
+      </BaseScreenComponent>
+    );
+  }
+
+  private handleGoBack = () =>
+    Alert.alert(
+      I18n.t("onboarding.alert.title"),
+      I18n.t("onboarding.alert.description"),
+      [
+        {
+          text: I18n.t("global.buttons.cancel"),
+          style: "cancel"
+        },
+        {
+          text: I18n.t("global.buttons.exit"),
+          style: "default",
+          onPress: () => this.props.dispatch(abortOnboarding())
```

Comment:
It's better to use **_mapDispatchToProps_** instead of dire
> encapsulating the dispatch logic into function makes the implementation more declarative.

Reply:
changed in ddc86001